app/assets/javascripts/discourse/app/components/topic-map/topic-map-link.gjs

```diff
@@ -0,0 +1,42 @@
+import Component from "@glimmer/component";
+import replaceEmoji from "discourse/helpers/replace-emoji";
+import and from "truth-helpers/helpers/and";
+
+export default class TopicMapLink extends Component {
+  get linkClasses() {
+    return this.args.attachment
+      ? "topic-link track-link attachment"
+      : "topic-link track-link";
+  }
+
+  get truncatedContent() {
+    const truncateLength = 85;
```

Comment:
Probably micro-optimizing, but might be worth extracting into a const outside of this `function` (even this `class`)?